rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -113,8 +116,14 @@ public static void startServer(Configuration conf, MetricRegistry metricsRegistr
   /**
    * Register desired metrics.
    */
-  private static void registerMetrics()
+  private static void registerMetrics(Configuration conf)
   {
+    if (CacheConfig.isOnMaster(conf)) {
+      StatsDReporter.forRegistry(metrics)
```

Comment:
Can you assert that values are not null ? What will happen if bad values are given ? Does it show up in logs ?

Reply:
Which values? For `metrics` and `conf`?

Reply:
if the value of host or port is wrong.

Reply:
These values are access through the configuration, so there will always be default values. If the host/port is invalid, StatsDReporter will log an error.